include/flatbuffers/flatbuffers.h

```diff
@@ -123,6 +125,15 @@
   #define FLATBUFFERS_NOEXCEPT
 #endif
 
+// NOTE: the FLATBUFFERS_DELETE_FUNC macro may change the access mode to
+// private, so be sure to put it at the end or reset access mode explicitly.
+#if (!defined(_MSC_VER) || _MSC_FULL_VER >= 180020827) && \
+    (!defined(__GNUC__) || (__GNUC__ * 100 + __GNUC_MINOR__ >= 404))
+  #define FLATBUFFERS_DELETE_FUNC(func) func = delete;
+#else
+  #define FLATBUFFERS_DELETE_FUNC(func) private: func;
```

Comment:
Yep, this is just doing the private declare for older compilers/stds. I think erring on the side of making things too private is better than pushing up to public by accident (in case the macro is used in a private/protected section), so let's leave it as is.